csrc/fusion_segmenter.cpp

```diff
@@ -1817,8 +1817,9 @@ void eraseInputDistinctRootDomains(Fusion* fusion) {
       if (tv->getLoopDomain() == tv->getAllocationDomain()) {
         new_loop = new_alloc;
       } else {
+        // scatter output wouldn't have matching loop and logical domain
         NVF_ERROR(
-            tv->getLoopDomain() == tv->getLogicalDomain(),
+            tv->getLoopDomain().size() == tv->getLogicalDomain().size(),
```

Comment:
I'd just remove this assertion entirely. The loop domain of a segment input shouldn't matter for anything.